clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureBaseClient.groovy

```diff
@@ -46,4 +57,53 @@ public abstract class AzureBaseClient {
       throw new RuntimeException("Failed to create Azure Resource Management Service", e)
     }
   }
+
+  public String getAzureRESTJson(AzureCredentials creds, String baseUrl, String targetUrl, List<String> queryParameters){
+    String result = null
+    String url = AzureUtilities.getAzureRESTUrl(subscriptionId, baseUrl, targetUrl, queryParameters)
+
+    try {
+      // create HTTP Client
+      def config = buildConfiguration(creds)
+      HttpClientBuilder httpClientBuilder = config.create(HttpClientBuilder.class)
+
+      String proxyHost = System.getProperty("http.proxyHost");
+      String proxyPort = System.getProperty("http.proxyPort");
+      if ((proxyHost != null) && (proxyPort != null)) {
+        HttpHost proxy = new HttpHost(proxyHost, Integer.parseInt(proxyPort));
+        if (proxy != null) {
+          httpClientBuilder.setProxy(proxy);
+        }
+      }
+
+      def httpClient = httpClientBuilder.build();
+
+      // Create HTTP transport objects
+      HttpGet httpRequest = new HttpGet(url);
+
+      // Set Headers
+      httpRequest.setHeader("Content-Type", "application/json");
+
+      // Send Request
+      HttpResponse httpResponse = httpClient.execute(httpRequest);
+      int statusCode = httpResponse.getStatusLine().getStatusCode();
+      if (statusCode != HttpStatus.SC_OK) {
+        ServiceException ex = ServiceException.createFromJson(httpRequest, null, httpResponse, httpResponse.getEntity());
+        throw ex;
+      }
+
+      // Deserialize Response
+      if (statusCode == HttpStatus.SC_OK) {
+        def entity = httpResponse.getEntity()
+        def responseContent = entity.getContent()
+        result = IOUtils.toString(responseContent)
+      }
+    }
+    catch (Exception e) {
+      log.info("getVMImagesAll -> Unexpected exception " + e.toString())
```

Comment:
Suspect that this is redundant given you're throwing an exception immediately afterwards)